core/action/action.go

```diff
@@ -131,6 +133,10 @@ func (action *Action) Execute(pair *models.RequestResponsePair) error {
 	//adding 200 ms to include some buffer for it to return response
 	time.Sleep(time.Duration(200+action.DelayInMs) * time.Millisecond)
 
+	journalID := <-journalIDChannel
```

Comment:
this is gonna be blocking, and the delay timing could be wrong.

Reply:
probably move this to the end of the execution?

Reply:
It won't be block because sending to channel will be instantaneous in another routine. I have put delay of 200ms before so that it waits for response to reach first followed by callback in the cases where delay is set to 0 explicitly by the user.

Reply:
but if the sender for whenever reason fails to send an id, the action goroutine is  gonna be waiting indefinitely, I think it's safer to move this receiver to the end of the action execution, and also set a timeout using the select statement, because the last thing we want to have is lots of stuck action goroutines :(

Reply:
done